WireAuthentication/Sources/WireAuthenticationLogic/SSO/SSOLinkGenerator.swift

```diff
@@ -0,0 +1,154 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireAuthenticationAPI
+
+package final class SSOLinkGenerator: SSOLinkGeneratorProtocol {
+
+    private let authenticationAPI: AuthenticationAPI
+    private let baseURL: URL
+    private let callbackScheme: String
+    private let defaults: UserDefaults
+
+    package init(
+        authenticationAPI: AuthenticationAPI,
+        baseURL: URL,
+        callbackScheme: String,
+        defaults: UserDefaults
+    ) {
+        self.authenticationAPI = authenticationAPI
+        self.baseURL = baseURL
+        self.callbackScheme = callbackScheme
+        self.defaults = defaults
+    }
+
+    package func generateSSOLink(ssoCode: UUID) async throws -> URL {
+        try await authenticationAPI.validateLoginToken(ssoCode: ssoCode)
+        return try await buildSSOLink(ssoCode: ssoCode)
+    }
+
+    package func flushToken() {
+        SSOLoginVerificationToken.flush(in: defaults)
+    }
+
+    /// Generate the link to the SSO authentication screen
+    ///
+    /// - Parameters:
+    ///   - baseURL: Backend URL.
+    ///   - ssoCode: SSO code
+    ///   - callbackScheme: The URL scheme that where the callback will be provided.
+    /// - Returns: URL to the SSO authentication screen
+    @MainActor
+    func buildSSOLink(ssoCode: UUID) async throws -> URL {
+        let validationToken = SSOLoginVerificationToken()
+        var components = URLComponents()
+        components.scheme = "https"
+        components.host = baseURL.host
+        components.path = "/sso/initiate-login/\(ssoCode.uuidString)"
```

Comment:
question: does this need to be api versioned?

Reply:
No, should be `HEAD https://prod-nginz-https.wire.com/sso/initiate-login/<IDP code>`
https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/18416063/Use+case+Initiate+SSO+login